indexer/services/ender/src/lib/kafka-publisher.ts

```diff
@@ -43,34 +55,135 @@ export class KafkaPublisher {
   }
 
   public addEvent(event: ConsolidatedKafkaEvent) {
-    switch (event.topic) {
+    this.getMessages(event.topic)!.push(event.message);
+  }
+
+  /**
+   * Helper function to get messages for a given topic.
+   *
+   * @param kafkaTopic
+   * @private
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private getMessages(kafkaTopic: KafkaTopics): any[] | undefined {
+    switch (kafkaTopic) {
       case KafkaTopics.TO_WEBSOCKETS_SUBACCOUNTS:
-        this.subaccountMessages.push(event.message);
-        break;
+        return this.subaccountMessages;
       case KafkaTopics.TO_WEBSOCKETS_TRADES:
-        this.tradeMessages.push(event.message);
-        break;
+        return this.tradeMessages;
       case KafkaTopics.TO_WEBSOCKETS_MARKETS:
-        this.marketMessages.push(event.message);
-        break;
+        return this.marketMessages;
       case KafkaTopics.TO_WEBSOCKETS_CANDLES:
-        this.candleMessages.push(event.message);
-        break;
+        return this.candleMessages;
       case KafkaTopics.TO_VULCAN:
-        this.vulcanMessages.push(event.message);
-        break;
+        return this.vulcanMessages;
       default:
         throw new Error('Invalid Topic');
     }
   }
 
   /**
-   * Sort trade events by block height, transaction index, and event index in ascending order,
-   * where the first trade event should be the earliest trade in the block.
+   * Extract fills from subaccount message contents.
+   *
+   * @param contents
+   * @private
+   */
+  private extractFills(contents: string): FillSubaccountMessageContents[] | undefined {
+    try {
+      const subaccountContents:
+      SubaccountMessageContents = JSON.parse(contents) as SubaccountMessageContents;
```

Comment:
nit: Let's add a TODO + linear ticket for adding a better way to aggregate messages. In addition to orders, we also want to make it so perpetual positions / asset positions only have a single update per block, so a more generic solution would be better. I think the current fix is fine for the short-term.
Having to do an extra JSON.parse per order message is un-ideal, and the JSON.stringify in the order handler is now redundant. It'd be better to add more generic logic to aggregate different types of subaccount messages rather than do post-processing.

Reply:
Actually, what do you think about updating the annotated messages such that it also contains a `contents` property? I think that's a small enough change that can be scoped into this fix, rather than waiting for the ticket to allow more generic aggregation of subaccount events.

Reply:
done